routing/missioncontrol.go

```diff
@@ -339,6 +339,48 @@ func (p *paymentSession) ReportChannelFailure(e uint64) {
 	p.mc.Unlock()
 }
 
+// ReportChannelFailureWithUpdate handles a failure message that contains a
+// channel update. It assume that an attempt has been made to apply the update
+// and only requires the boolean result of this as an input parameter.
+//
+// If the update failed, it prunes the edge. Regardless of whether the update
+// succeeded or failed, it marks the edge as 'failed once'. The next time it
+// fails, the whole node will be pruned.
+//
+// TODO(joostjager): Base record keeping on a tuple consisting of both endpoints
+// of the channel instead of the channel id. It may be that a remote node
+// chooses another channel to route the payment over and there is currently no
+// way for us to determine the used channel reliably.
+func (p *paymentSession) ReportChannelFailureWithUpdate(
+	errSource Vertex, failedChanID uint64, success bool) {
+
+	// If the update could not be applied, prune the edge. There is no
+	// reason to continue trying this channel.
+	if !success {
+		p.ReportChannelFailure(failedChanID)
+	}
+
+	// We'll now check to see if we've already
```

Comment:
nit: lines can be made longer in this method.

Reply:
rewrapped.